apps/public/src/app/(general)/samenvatting/page.tsx

```diff
@@ -21,64 +19,33 @@ export const generateMetadata = async () => {
 }
 
 export default async () => {
-  // Get session variables from cookies
   const cookieStore = await cookies()
   const meldingId = cookieStore.get('id')?.value
   const token = cookieStore.get('token')?.value
   const location = cookieStore.get('location')?.value
 
-  if (!meldingId || !token) return undefined
-
-  // Primary form
-
-  const { data: staticFormsData, error: staticFormsError } = await getStaticForm()
-
-  if (staticFormsError) return handleApiError(staticFormsError)
-
-  const primaryFormId = staticFormsData?.find((form) => form.type === 'primary')?.id
-
-  if (!primaryFormId) return 'Primary form id not found'
-
-  const { data: primaryFormData, error: primaryFormError } = await getStaticFormByStaticFormId({
-    path: { static_form_id: primaryFormId },
-  })
-
-  if (primaryFormError) return handleApiError(primaryFormError)
-
-  if (!primaryFormData) return 'Primary form data not found'
-
-  const primaryForm = primaryFormData.components[0]
-
-  // Melding
-
-  const { data: meldingData, error: meldingError } = await getMeldingByMeldingIdMelder({
-    path: { melding_id: parseInt(meldingId, 10) },
-    query: { token },
-  })
-
-  if (meldingError) return handleApiError(meldingError)
-
-  if (!meldingData) return 'Melding data not found'
+  const t = await getTranslations()
 
-  // Additional questions
+  if (!meldingId || !token) throw new Error('Could not retrieve meldingId or token')
 
-  const { data: additionalQuestionsData, error: additionalQuestionsErrors } = await getMeldingByMeldingIdAnswers({
-    path: { melding_id: parseInt(meldingId, 10) },
-    query: { token },
-  })
+  const meldingData = await getMeldingData(meldingId, token)
+  if (typeof meldingData === 'string') return meldingData
```

Comment:
Maybe we can use the same pattern here as our API client functions. So `getMeldingData` returns data or an error. On error, return error message. That's a bit more explicit than `typeof data === 'string'`

Reply:
I'm not sure what you mean here. I tried this for the `getPrimaryFormSummary` function and it works as expected. What is the TypeScript error you run into?